test/upsert/mzcompose.py

```diff
@@ -102,3 +126,55 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
             ci_util.upload_junit_report(
                 "testdrive", Path(__file__).parent / junit_report
             )
+
+
+def workflow_rehydration(c: Composition) -> None:
+    """Test creating sources in a remote clusterd process."""
+
+    c.down(destroy_volumes=True)
+
+    dependencies = [
+        "materialized",
+        "zookeeper",
+        "kafka",
+        "schema-registry",
+        "clusterd1",
+    ]
+
+    c.up("materialized")
+    c.run("testdrive", "rehydration/01-setup.td")
+
+    for (style, mz) in [
+        (
+            "with DISK",
+            Materialized(
+                options=[
+                    "--orchestrator-process-scratch-directory=/mzdata/source_data",
+                ],
+                additional_system_parameter_defaults=[
+                    "upsert_source_disk_default=true"
+                ],
+            ),
+        ),
+        ("without DISK", Materialized()),
+    ]:
+
+        with c.override(
+            mz,
+            Testdrive(no_reset=True, consistent_seed=True),
+        ):
+            print(f"Running rehydration workflow {style}")
+
+            c.up(*dependencies)
+
+            c.run("testdrive", "rehydration/02-source-setup.td")
+
+            c.kill("materialized")
+            c.kill("clusterd1")
+            c.up("materialized")
+            c.up("clusterd1")
+
+            c.run("testdrive", "rehydration/03-after-rehydration.td")
+
+        c.run("testdrive", "rehydration/04-reset.td")
```

Comment:
Is the reset needed? It's empty

Reply:
Its the easiest way to get `testdrive` to run all the reset logic!

Reply:
TIL! (I will try in my branch too, I was having trouble with resetting things)